### PR TITLE
solution

### PR DESCRIPTION
**Pull request by abcdesign1**
[DEMO LINK](https://vadym4che.github.io/react_person/)

### CODE REVIEW

File: src/components/Person/Person.jsx [hunk 1]
```diff
@@ -1 +1,32 @@
...
+        <p className="Person__age">
+          {'I am '}
+          {age}
+        </p>
+        )}
+      <p className="Person__partner">
+        { isMarried
+          ? `${partnerName} is my ${sex === 'm' ? 'wife' : 'husband'}`
+          : 'I am not married'}
+      </p>
```
Comment (YaroslavDzyadiv) [inline src/components/Person/Person.jsx:29]:
```suggestion
      <h2 className="Person__name">
        {'My name is '}
        {name}
      </h2>
      
      { age
        && (
        <p className="Person__age">
          {'I am '}
          {age}
        </p>
        )}
        
      <p className="Person__partner">
        { isMarried
          ? `${partnerName} is my ${sex === 'm' ? 'wife' : 'husband'}`
          : 'I am not married'}
      </p>
```

Comment (YaroslavDzyadiv) [reply src/components/Person/Person.jsx:29]: I think you should add spaces between blocks to improve code readability.

**Review by YaroslavDzyadiv — APPROVED** [review]
Good job!

File: src/components/Person/Person.jsx [hunk 2]
```diff
@@ -1 +1,32 @@
...
+    sex,
+    isMarried,
+    partnerName,
+  } = person;
+
+  return (
+    <section className="Person">
+      <h2 className="Person__name">
+        {'My name is '}
+        {name}
```
Comment (illiarerun) [inline src/components/Person/Person.jsx:16]: use template strings - `` to improve readability

File: src/components/Person/Person.jsx [hunk 3]
```diff
@@ -1 +1,32 @@
...
+      </h2>
+      { age
+        && (
+        <p className="Person__age">
+          {'I am '}
+          {age}
```
Comment (illiarerun) [inline src/components/Person/Person.jsx:22]: same here

File: src/components/Person/Person.jsx [hunk 4]
```diff
@@ -1 +1,32 @@
...
+        </p>
+        )}
```
Comment (illiarerun) [inline src/components/Person/Person.jsx:24]: add indents for nested elements and fix your formatting
```suggestion
      {age && (
        <p className="Person__age">
          {'I am '}
          {age}
        </p>
      )}
```

File: src/components/Person/Person.jsx (hunk 1 above)
Comment (illiarerun) [reply src/components/Person/Person.jsx:29]: Agree about spaces

File: src/components/Person/Person.jsx [hunk 5]
```diff
@@ -1 +1,32 @@
...
+      <p className="Person__partner">
+        { isMarried
+          ? `${partnerName} is my ${sex === 'm' ? 'wife' : 'husband'}`
```
Comment (illiarerun) [inline src/components/Person/Person.jsx:27]: don't use nested ternaries, create a variable for `wife/husband`

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/components/Person/Person.jsx (hunk 1 above)
Comment (vadym4che) [reply src/components/Person/Person.jsx]: fixed all

File: src/components/Person/Person.jsx (hunk 2 above)
Comment (vadym4che) [reply src/components/Person/Person.jsx]: fixed all

File: src/components/Person/Person.jsx (hunk 3 above)
Comment (vadym4che) [reply src/components/Person/Person.jsx]: fixed all

File: src/components/Person/Person.jsx (hunk 4 above)
Comment (vadym4che) [reply src/components/Person/Person.jsx]: fixed all

File: src/components/Person/Person.jsx (hunk 5 above)
Comment (vadym4che) [reply src/components/Person/Person.jsx]: fixed all

File: src/components/Person/Person.jsx [hunk 6]
```diff
@@ -1 +1,32 @@
...
+    ? 'husband'
+    : 'wife';
+
+  const married = isMarried
+    ? `${partnerName} is my ${partner}`
+    : 'I am not married';
+
+  return (
+    <section className="Person">
+      <h2 className="Person__name">{`My name is ${name}`}</h2>
```
Comment (sergii-nosachenko) [inline src/components/Person/Person.jsx:23]:
```suggestion
      <h2 className="Person__name">
        {`My name is ${name}`}
      </h2>
```

**Review by sergii-nosachenko — APPROVED** [review]

**Pull request closed without merging**